src/svm/utils/events.ts

```diff
@@ -0,0 +1,94 @@
+import { BN } from "@coral-xyz/anchor";
+import web3 from "@solana/web3-v2.js";
+import {
+  BridgedToHubPoolEvent,
+  ClaimedRelayerRefundEvent,
+  EmergencyDeletedRootBundleEvent,
+  EnabledDepositRouteEvent,
+  EventData,
+  EventName,
+  ExecutedRelayerRefundRootEvent,
+  FilledRelayEvent,
+  FundsDepositedEvent,
+  PausedDepositsEvent,
+  PausedFillsEvent,
+  RelayedRootBundleEvent,
+  RequestedSlowFillEvent,
+  SetXDomainAdminEvent,
+  SVMEventNames,
+  TokensBridgedEvent,
+  TransferredOwnershipEvent,
+} from "../types";
+
+/**
+ * Parses event data from a transaction.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function parseEventData(eventData: any): any {
+  if (!eventData) return eventData;
+
+  if (Array.isArray(eventData)) {
+    return eventData.map(parseEventData);
+  }
+
+  if (typeof eventData === "object") {
+    if (eventData.constructor.name === "PublicKey") {
+      return web3.address(eventData.toString());
+    }
+    if (BN.isBN(eventData)) {
+      return BigInt(eventData.toString());
+    }
+
+    return Object.fromEntries(Object.entries(eventData).map(([key, value]) => [key, parseEventData(value)]));
+  }
+
+  return eventData;
+}
+
+/**
+ * Gets the event name from a raw name.
+ */
+export function getEventName(rawName?: string): EventName {
+  if (!rawName) throw new Error("Raw name is undefined");
```

Comment:
This should never happen to we throw an error

Reply:
why accept optional param in the first place?